### PR TITLE
Added accessibility attributes to popover

### LINKED ISSUE #3149: Update @yamada-ui/popover accessibility
### Description

This issue aims to address accessibility deficiencies in our popover component as per W3C guidelines.
The component currently does not meet the accessibility standards outlined in the W3C documentation.

The goal is to ensure the component is accessible to all users, including those using assistive technologies.
Ensure that all unchecked items in this checklist should be addressed.

### Problem Statement/Justification

improve accessibility

### Proposed Solution or API

Add the following roles and attributes:

- aria-haspopup
- aria-expanded
- role="dialog"
- aria-labelledby
- aria-describedby
- aria-hidden

### Alternatives

_No response_

### Are you willing to participate in implementing this feature and create a pull request include the implementation?

Yes

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by kaehehehe**
Closes #3149

## Description

Added accessibility attributes to popover.

## Is this a breaking change (Yes/No):

No

### CODE REVIEW

**Comment by chida09** [comment]
@kaehehehe 
The `Quality / Test` has failed, so please check.
https://github.com/yamada-ui/yamada-ui/actions/runs/11417571465/job/31770049437?pr=3162

**Comment by kaehehehe** [comment]
@chida09 
Yes, I know. I'm currently checking

**Comment by kaehehehe** [comment]
fixed! @chida09

**Comment by kaehehehe** [comment]
I want to add `aria-expanded` to `PopoverTrigger`, but doing so causes the test to fail. Specifying `role="button"` allows the test to pass, but since the children of PopoverTrigger are already buttons, this attribute seems unnecessary.
How can we resolve this issue?

스크린샷 2024-10-20 13 07 53

File: packages/components/popover/src/popover-content.tsx [hunk 1]
```diff
@@ -138,9 +140,13 @@ export const PopoverContent = motionForwardRef<PopoverContentProps, "section">(
             ? getPopoverContentProps(animation, duration)
             : {})}
           {...getPopoverProps(rest, ref)}
+          aria-describedby={describedbyId}
```
Comment (hirotomoyamada) [inline packages/components/popover/src/popover-content.tsx:143]: Please move `aria-describedby={describedbyId}` to `getPopoverProps`.

File: packages/components/popover/src/popover-content.tsx [hunk 2]
```diff
@@ -138,9 +140,13 @@ export const PopoverContent = motionForwardRef<PopoverContentProps, "section">(
...
+          aria-hidden={!isOpen}
```
Comment (hirotomoyamada) [inline packages/components/popover/src/popover-content.tsx:144]: Please move `aria-hidden={!isOpen}` to `getPopoverProps`.

File: packages/components/popover/src/popover-content.tsx [hunk 3]
```diff
@@ -138,9 +140,13 @@ export const PopoverContent = motionForwardRef<PopoverContentProps, "section">(
...
+          aria-labelledby={labelledbyId}
```
Comment (hirotomoyamada) [inline packages/components/popover/src/popover-content.tsx:145]: Please move `aria-labelledby={labelledbyId}` to `getPopoverProps`.

File: packages/components/popover/src/popover-content.tsx [hunk 4]
```diff
@@ -138,9 +140,13 @@ export const PopoverContent = motionForwardRef<PopoverContentProps, "section">(
...
           animate={isOpen ? "enter" : "exit"}
           exit="exit"
           initial="exit"
+          role="dialog"
```
Comment (hirotomoyamada) [inline packages/components/popover/src/popover-content.tsx:149]: Please move `role="dialog"` to `getPopoverProps`.

File: packages/components/popover/src/popover-trigger.tsx [hunk 5]
```diff
@@ -9,7 +9,14 @@ export const PopoverTrigger: FC<PropsWithChildren<{}>> = ({ children }) => {
   } & ReactElement
   const { getTriggerProps } = usePopover()
 
-  return cloneElement(child, getTriggerProps(child.props, child.ref))
+  const triggerProps = getTriggerProps(child.props, child.ref)
+
+  const enhancedTriggerProps = {
+    ...triggerProps,
+    "aria-haspopup": "true",
```
Comment (hirotomoyamada) [inline packages/components/popover/src/popover-trigger.tsx:16]: Please move `"aria-haspopup": "true"` to `getTriggerProps`.

File: packages/components/popover/src/popover.tsx
Comment (hirotomoyamada) [inline packages/components/popover/src/popover.tsx:1]: Set `getPopoverProps` to the trigger element's `id` for `aria-controls`.

**Review by hirotomoyamada — DISMISSED** [review]

**Review by hirotomoyamada — DISMISSED** [review]

**Review by hirotomoyamada — APPROVED** [review]

**Pull request merged**